docs/proposals/data-cache.md

```diff
@@ -0,0 +1,160 @@
+2. Add a handler which primary purpose is to parse parameters about the data cache in user-submitted jobs and provide default values for some optional fields.
+3. Add a checker that checks to see if available fluids are deployed within the cluster and, if not, prompts the user that data cache is not available.
+4. Add a client that interacts with the api server. After the parameters related to the data cache are resolved, the parameters are submitted to the fluid in the cluster through the api server and the dataset and create the dataset and cache runtime (usually alluxio runtime), which are apis provided by the fluid.
+5. Add a watch feature that will use client to monitor if a new pv&pvc pair are created in the cluster (if dataset and runtime are successfully created, fluid will bind them and create a pair of pv and pvc).
+6. Add a Injector which injects the pvc created by fluid into the container. This allows jobs to access files in the cached path and enjoy data acceleration.
+
+## Implementation
+
+### Sequence Diagram
+
+FYR. The final implementation may be slightly different from this diagram.
+
+![](../img/data_cache_sequence.png)
+The general process is as follows:
+
+1. The user submits a yaml file describing the job whose field contains the parameters of the data cache, which means that the job will use the feature of the data cache.
+2. KubeDL's Job Controller detects that a job with data cache parameters has been created, and it passes the parameters to the handler responsible for enable data cache.
+3. After verifying that the parameters are correct, the cache handler adds defaults to the missing parameters and creates a client. The client checks to see if Fluid has been deployed in the Kubernetes cluster, and if so, it will create the dataset and cache runtime by the api server.
+4. The volume controller from Fluid detects that there is a pair of dataset and runtime in the cluster that can be bound, and it will automatically create pv and assign pvc to it. Users can enjoy data cache by mounting this pvc.
+5. When the listener in KubeDL detects that pvc is created, it will notify the cache handler and pass relevant information about pvc to the handler.
+6. Handler receives the parameters of the pvc and injects the pvc into the container of the KubeDL job.
+
+With the above process, the dataset in the specified file system can be cached, which means that the time for the KubeDL job to access the dataset will be greatly reduced.
+
+### Cache Configuration
+
+Reference to the configuration of `modelVersion` in `example/tf/tf_job_mnist_modelversion.yaml`
+
+```yaml
+    apiVersion: "training.kubedl.io/v1alpha1"
+    kind: "TFJob"
+    metadata:
+      name: "mnist"
+      namespace: kubedl
+    spec:
+      cleanPodPolicy: None
++     fluidConfig:
```

Comment:
Another way to look at the configs is, we can separate the user facing configs and a central admin side configs, of 
course, ​user facing configs can provide options to override admin configs.

configs  like 'quota' probably should not be a per job config ?

Reply:
1. Actually, I'm a little confused about the "central admin side configs". Does this config fill in some default values for users? If so, in the list below example configuration, I've given some optional arguments. It means that the user does not have to provide these optional parameters, and they will be filled in automatically by kubeDL. If not, can you be a little more specific?
2. Quota is a required parameter to generate the cache runtime. If we do not configure one quota for each job, this means that multiple datasets will share the same cache runtime, and it doesn't seem feasible.